shared_model/builders/protobuf/transaction.hpp

```diff
@@ -0,0 +1,183 @@
+
+#include "backend/protobuf/transaction.hpp"
+
+#include "block.pb.h"
+#include "commands.pb.h"
+
+#include "amount/amount.hpp"
+#include "builders/protobuf/helpers.hpp"
+#include "interfaces/common_objects/types.hpp"
+
+namespace shared_model {
+  namespace proto {
+    template <int S = 0>
+    class TemplateTransactionBuilder {
+     private:
+      template <int>
+      friend class TemplateTransactionBuilder;
+
+      enum RequiredFields { Command, CreatorAccountId, TxCounter, TOTAL };
+
+      template <int s>
+      using NextBuilder = TemplateTransactionBuilder<S | (1 << s)>;
+
+      iroha::protocol::Transaction transaction_;
+
+      template <int Sp>
+      TemplateTransactionBuilder(const TemplateTransactionBuilder<Sp> &o)
+          : transaction_(o.transaction_) {}
+
+     public:
+      TemplateTransactionBuilder() = default;
+
+      NextBuilder<CreatorAccountId> creatorAccountId(
+          const interface::types::AccountIdType &account_id) {
+        transaction_.mutable_payload()->set_creator_account_id(account_id);
+        return *this;
+      }
+
+      NextBuilder<TxCounter> txCounter(Transaction::TxCounterType tx_counter) {
+        transaction_.mutable_payload()->set_tx_counter(tx_counter);
+        return *this;
+      }
+
+      NextBuilder<Command> addAssetQuantity(
+          const interface::types::AccountIdType &account_id,
+          const interface::types::AssetIdType &asset_id,
+          const std::string &amount) {
+        auto command = transaction_.mutable_payload()
+                           ->add_commands()
+                           ->mutable_add_asset_quantity();
+        command->set_account_id(account_id);
+        command->set_asset_id(asset_id);
+        addAmount(command->mutable_amount(), amount);
+        return *this;
+      }
+
+      NextBuilder<Command> addAddPeer(
+          const interface::types::AddressType &address,
+          const interface::types::PubkeyType &peer_key) {
+        auto command =
+            transaction_.mutable_payload()->add_commands()->mutable_add_peer();
+        command->set_address(address);
+        command->set_peer_key(peer_key.blob());
+        return *this;
+      }
+
+      NextBuilder<Command> addAddSignatory(
+          const interface::types::AddressType &account_id,
+          const interface::types::PubkeyType &public_key) {
+        auto command = transaction_.mutable_payload()
+                           ->add_commands()
+                           ->mutable_add_signatory();
+        command->set_account_id(account_id);
+        command->set_public_key(public_key.blob());
+        return *this;
+      }
+
+      NextBuilder<Command> addRemoveSignatory(
+          const interface::types::AddressType &account_id,
+          const interface::types::PubkeyType &public_key) {
+        auto command = transaction_.mutable_payload()
+                           ->add_commands()
+                           ->mutable_remove_sign();
```

Comment:
Is it exact name of `RemoveSignatory` in generated cpp file? I feel that name should be `remove_signatory`.

Reply:
Yeah it is correct :( proto generates some weird names